nimble/host/services/asha/src/ble_svc_asha.c

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <assert.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "sysinit/sysinit.h"
+#include "bsp/bsp.h"
+#include "host/ble_hs.h"
+#include "host/ble_uuid.h"
+#include "services/asha/ble_svc_asha.h"
+
+
+static uint8_t audio_status;
+static struct ble_svc_asha_audio_control audio_control_point;
+static uint8_t read_only_buffer[READ_ONLY_BUFFER_SIZE] = { 0x01,DEVICE_SIDE,
+        0x31, 0x00,0x01, 0x02, 0x03, 0x04, 0x05, 0x06,
+        0x00, 0x00, 0x00, 0x07, 0x00, 0x07,0x00 };
+static uint16_t psm_handle = 0x0080;
+static uint16_t audio_status_handle;
+
+static int
+read_device_prop(uint16_t conn_handle, uint16_t attr_handle,
+        struct ble_gatt_access_ctxt *ctxt, void *arg);
+static int
+control_audio(uint16_t conn_handle, uint16_t attr_handle,
+        struct ble_gatt_access_ctxt *ctxt, void *arg);
+static int
+read_audio_status(uint16_t conn_handle, uint16_t attr_handle,
+        struct ble_gatt_access_ctxt *ctxt, void *arg);
+static int
+set_volume(uint16_t conn_handle, uint16_t attr_handle,
+        struct ble_gatt_access_ctxt *ctxt, void *arg);
+static int
+read_l2cap_chan_psm(uint16_t conn_handle, uint16_t attr_handle,
+        struct ble_gatt_access_ctxt *ctxt, void *arg);
+
+static const struct ble_gatt_svc_def ble_svc_asha[] = {
+        {
+        /* Service: HEARING AID. */
+            .type = BLE_GATT_SVC_TYPE_PRIMARY,
+            .uuid = BLE_UUID16_DECLARE(BLE_SVC_ASHA_UUID16),
+            .characteristics = (struct ble_gatt_chr_def[] ){ {
+        /* Characteristic: read_device_prop. */
+                .uuid = BLE_UUID128_DECLARE(
+                    BLE_SVC_ASHA_CHR_UUID128_DEVICE_PROP),
+                .access_cb = read_device_prop,
+                .flags = BLE_GATT_CHR_F_READ,
+            },{
+                /* Characteristic: control_audior. */
+                .uuid = BLE_UUID128_DECLARE(
+                    BLE_SVC_ASHA_CHR_UUID128_CONTROL_AUDIO),
+                .access_cb =control_audio,
```

Comment:
whitespaces around `=` - there are quite a lot of such coding style issues so please review complete code and fix